src/com/mendix/widget/RangeSlider/components/RangeSlider.ts

```diff
@@ -0,0 +1,150 @@
+        );
+    }
+
+    private calculateMarks(props: SliderProps): Marks {
+        const marks: Marks = {};
+        if (this.isValidMinMax(props) && props.noOfMarkers >= 2) {
+            const interval = (props.maxValue - props.minValue) / (props.noOfMarkers - 1);
+            for (let i = 0; i < props.noOfMarkers; i++) {
+                const value = parseFloat((props.minValue + (i * interval)).toFixed(props.decimalPlaces));
+                marks[value] = value;
+            }
+        }
+        return marks;
+    }
+
+    private isValidMinMax(props: SliderProps): boolean {
+        const { maxValue, minValue } = props;
+        return typeof maxValue === "number" && typeof minValue === "number" && minValue < maxValue;
+    }
+
+    private validateSettings(props: SliderProps): string {
+        const message: string[] = [];
+        const validMax = typeof props.maxValue === "number";
+        const validMin = typeof props.minValue === "number";
+        if (!validMax) {
+            message.push("Maximum value is required");
+        }
+        if (!validMin) {
+            message.push("Minimum value is required");
+        }
+        if (typeof props.lowerBound !== "number") {
+            message.push("Lower bound value is required");
+        }
+        if (typeof props.upperBound !== "number") {
+            message.push("Upper bound value is required");
+        }
+        if (validMin && validMax && (props.minValue >= props.maxValue)) {
+            message.push(`Minimum value ${props.minValue} should be less than the maximum value ${props.maxValue}`);
+        }
+        if (!props.stepValue || props.stepValue <= 0) {
+            message.push(`Step value ${props.stepValue} should be greater than 0`);
+        } else if (validMax && validMin && (props.maxValue - props.minValue) % props.stepValue > 0) {
+            message.push(`Step value is invalid, max - min (${props.maxValue} - ${props.minValue}) 
+            should be evenly divisible by the step value ${props.stepValue}`);
+        }
+
+        return message.join(", ");
+    }
+
+    private validateValues(props: SliderProps): string {
+        const message: string[] = [];
+        if (props.lowerBound > props.maxValue) {
+            message.push(`Lower bound ${props.lowerBound} should be less than the maximum ${props.maxValue}`);
+        }
+        if (props.lowerBound < props.minValue) {
+            message.push(`Lower bound ${props.lowerBound} should be greater than the minimum ${props.minValue}`);
+        }
+        if (props.upperBound > props.maxValue) {
+            message.push(`Upper bound ${props.upperBound} should be less than the maximum ${props.maxValue}`);
+        }
+        if (props.upperBound < props.minValue) {
+            message.push(`Upper bound ${props.upperBound} should be greater than the minimum ${props.minValue}`);
+        }
+        return message.join(", ");
+    }
+
+    private getTooltipText(value: number): string {
```

Comment:
value will never be bigger than max, or smaller than min.
Should we keep the original props.value instead? else it will be displaying falsely the the max or min value